src/analysis/asymptotics/specific_asymptotics.lean

```diff
@@ -98,3 +98,83 @@ begin
 end
 
 end normed_linear_ordered_field
+
+section real
+
+open_locale big_operators
+open finset
+
+lemma asymptotics.is_o.sum_range {α : Type*} [normed_group α]
+  {f : ℕ → α} {g : ℕ → ℝ} (h : is_o f g at_top)
+  (hg : 0 ≤ g) (h'g : tendsto (λ n, ∑ i in range n, g i) at_top at_top) :
+  is_o (λ n, ∑ i in range n, f i) (λ n, ∑ i in range n, g i) at_top :=
+begin
+  have A : ∀ i, ∥g i∥ = g i := λ i, real.norm_of_nonneg (hg i),
+  have B : ∀ n, ∥∑ i in range n, g i∥ = ∑ i in range n, g i,
+  { assume n,
+    rw [real.norm_eq_abs, abs_sum_of_nonneg'],
+    exact hg },
+  apply is_o_iff.2 (λ ε εpos, _),
+  obtain ⟨N, hN⟩ : ∃ (N : ℕ), ∀ (b : ℕ), N ≤ b → ∥f b∥ ≤ ε / 2 * g b,
+    by simpa only [A, eventually_at_top] using is_o_iff.mp h (half_pos εpos),
+  have : is_o (λ (n : ℕ), ∑ i in range N, f i) (λ (n : ℕ), ∑ i in range n, g i) at_top,
+  { apply is_o_const_left.2,
+    exact or.inr (h'g.congr (λ n, (B n).symm)) },
+  filter_upwards [is_o_iff.1 this (half_pos εpos), Ici_mem_at_top N] with n hn Nn,
+  calc ∥∑ i in range n, f i∥
+  = ∥∑ i in range N, f i + ∑ i in Ico N n, f i∥ :
+    by rw sum_range_add_sum_Ico _ Nn
+  ... ≤ ∥∑ i in range N, f i∥ + ∥∑ i in Ico N n, f i∥ :
+    norm_add_le _ _
+  ... ≤ ∥∑ i in range N, f i∥ + ∑ i in Ico N n, (ε / 2) * g i :
+    add_le_add le_rfl (norm_sum_le_of_le _ (λ i hi, hN _ (mem_Ico.1 hi).1))
+  ... ≤ ∥∑ i in range N, f i∥ + ∑ i in range n, (ε / 2) * g i :
+    begin
+      refine add_le_add le_rfl _,
+      apply sum_le_sum_of_subset_of_nonneg,
+      { rw range_eq_Ico,
+        exact Ico_subset_Ico (zero_le _) le_rfl },
+      { assume i hi hident,
+        exact mul_nonneg (half_pos εpos).le (hg i) }
+    end
+  ... ≤ (ε / 2) * ∥∑ i in range n, g i∥ + (ε / 2) * (∑ i in range n, g i) :
+    begin
+      rw ← mul_sum,
+      exact add_le_add hn (mul_le_mul_of_nonneg_left le_rfl (half_pos εpos).le),
+    end
+  ... = ε * ∥(∑ i in range n, g i)∥ : by { simp [B], ring }
```

Comment:
This suggestion might not be canonical style, but I find calc-blocks easier to read without `begin ... end` blocks in the middle. 

```suggestion
  calc ∥∑ i in range n, f i∥
      = ∥∑ i in range N, f i + ∑ i in Ico N n, f i∥                       : _
  ... ≤ ∥∑ i in range N, f i∥ + ∥∑ i in Ico N n, f i∥                     : norm_add_le _ _
  ... ≤ ∥∑ i in range N, f i∥ + ∑ i in Ico N n, (ε / 2) * g i             : _
  ... ≤ ∥∑ i in range N, f i∥ + ∑ i in range n, (ε / 2) * g i             : _
  ... ≤ (ε / 2) * ∥∑ i in range n, g i∥ + (ε / 2) * (∑ i in range n, g i) : _
  ... = ε * ∥(∑ i in range n, g i)∥                                       : by { simp [B], ring },
  { rw sum_range_add_sum_Ico _ Nn, },
  { exact add_le_add le_rfl (norm_sum_le_of_le _ (λ i hi, hN _ (mem_Ico.1 hi).1)), },
  { refine add_le_add le_rfl (sum_le_sum_of_subset_of_nonneg _ _),
    { rw range_eq_Ico,
      exact Ico_subset_Ico (zero_le _) le_rfl },
    { exact λ i hi hident, mul_nonneg (half_pos εpos).le (hg i) }, },
  { rw ← mul_sum,
    exact add_le_add hn (mul_le_mul_of_nonneg_left le_rfl (half_pos εpos).le), },
```

Reply:
I left two trivial suggestions (safe to ignore).

This looks good to me and I'm looking forward to the strong LLN!

Reply:
It's a matter of taste, but for the calc blocks I really prefer when proof obligations are dealt with in `begin...end` blocks, as otherwise I find it hard to see which proof corresponds to which block.